### PR TITLE
Providers api completed for resources.json swagger

### PR DESCRIPTION
**Pull request by shreyanshn**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json [hunk 1]
```diff
@@ -4573,14 +4573,38 @@
...
           "description": "The namespace of the resource provider."
         },
+        "authorizations": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceProviderAuthorization"
+          },
+          "description": "The authorizations fields of the resource provider."
+        },
+        "metadata": {
```
Comment (raych1) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json:4583]: What's this metadata? Does it have any required property?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json [hunk 2]
```diff
@@ -4573,14 +4573,38 @@
...
+          "type": "object",
+          "description": "The metadata."
+        },
         "registrationState": {
           "readOnly": true,
           "type": "string",
+          "enum": [
```
Comment (raych1) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json:4590]: Adding enum values will be breaking change.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json (hunk 1 above)
Comment (shreyanshn) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json:4583]: It's the metadata for the Resource Provider, if any. No required properties.

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json (hunk 2 above)
Comment (shreyanshn) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json:4590]: The enum values were missing earlier. I have to add them.

**Comment by raych1** [comment]
@Tiano2017 , can you please review this PR?

File: specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json [hunk 3]
```diff
@@ -4594,6 +4618,56 @@
...
+          "description": "The role definition id."
+        },
+        "managedByRoleDefinitionId": {
+          "type": "string",
+          "description": "The managed by role definition id."
+        },
+        "managedByAuthorization": {
+          "$ref": "#/definitions/ResourceProviderManagedByAuthorization"
+        },
+        "allowedThirdPartyExtensions": {
```
Comment (Tiano2017) [inline specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json:4638]: do we actually return these properties today?

Comment (shreyanshn) [reply specification/resources/resource-manager/Microsoft.Resources/stable/2019-10-01/resources.json]: No, we don't. Fixed.

**Comment by raych1** [comment]
@shreyanshn , removing properties is breaking change, what's the purpose to remove them?

**Comment by shreyanshn** [comment]
Like @Tiano2017 asked, we do not send these properties. @Tiano2017, can you please review again?

**Comment by raych1** [comment]
@Tiano2017 , could you please review the change again?

**Comment by Tiano2017** [comment]
@raych1 the PR looks good to me. It's trying to reflect what we have at service side. Is there any open issue you want to close on?

**Comment by raych1** [comment]
Hi @shreyanshn , as identified in the automated checks there are breaking changes, Please follow Azure Breaking changes approval request form as defined at [breaking change process](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37673/Retirements-and-Breaking-Changes). Please contact azureapirbcore@microsoft.com to get reviewed.

**Comment by raych1** [comment]
@shreyanshn , are you still working on this PR? Do you communicate with azure API board team in terms of breaking change?

**Comment by raych1** [comment]
Hi @shreyanshn , I will close this PR in 2 days if there's no response. Feel free to re-open this PR if necessary.

**Comment by raych1** [comment]
I'm closing this PR as there's no response.

**Pull request closed without merging**